ShopAppTests/App/Main/Views/Cells/CollectionView/ProductOptionsCollectionViewCellSpec.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  ProductOptionsCollectionViewCellSpec.swift
+//  ShopAppTests
+//
+//  Created by Evgeniy Antonov on 3/21/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import Quick
+import Nimble
+import ShopApp_Gateway
+
+@testable import ShopApp
+
+class ProductOptionsCollectionViewCellSpec: QuickSpec {
+    override func spec() {
+        var cell: ProductOptionsCollectionViewCell!
+        var collectionView: UICollectionView!
+        
+        beforeEach {
+            let provider = ProductOptionsCollectionProvider()
+            
+            let controllerCollectionView = UICollectionView(frame: CGRect.zero, collectionViewLayout: UICollectionViewFlowLayout())
+            controllerCollectionView.registerNibForCell(ProductOptionsCollectionViewCell.self)
+            
+            controllerCollectionView.dataSource = provider
+            controllerCollectionView.delegate = provider
+            
+            let option = ProductOption()
+            option.name = "Option name"
+            option.values = ["Option value 1", "Option value 2"]
+            provider.options = [option]
+            controllerCollectionView.reloadData()
+            
+            let indexPath = IndexPath(row: 0, section: 0)
+            let dequeuedCell: ProductOptionsCollectionViewCell = controllerCollectionView.dequeueReusableCellForIndexPath(indexPath)
+            cell = dequeuedCell
+            
+            collectionView = self.findView(withAccessibilityLabel: "collectionView", in: cell) as! UICollectionView
+        }
+        
+        describe("when cell configured") {
+            var values: [String]!
+            
+            beforeEach {
+                values = ["Value 1", "Value 2"]
+                let selectedValue = "Value 1"
+                cell.configure(with: values, selectedValue: selectedValue, delegate: nil)
+            }
+            
+            it("should have correct sections count") {
+                expect(collectionView.numberOfSections) == 1
+            }
+            
+            it("should have", closure: {
+                expect(collectionView.numberOfItems(inSection: 0)) == values.count
+            })
+        }
+        
+        describe("when option value did select") {
+            var values: [String]!
+            var delegateMock: ProductOptionsCollectionCellDelegateMock!
+            let selectedValue = "Value 1"
+            
+            beforeEach {
+                values = ["Value 1", "Value 2"]
+                delegateMock = ProductOptionsCollectionCellDelegateMock()
+                cell.configure(with: values, selectedValue: selectedValue, delegate: delegateMock)
```

Comment:
Can we select cell of uicollectionview instead calling method?

Reply:
In this method we need to check that method work correct and values passed to delegate. Does it have sense what object would call this method?

Reply:
We do it in this way all the time if it possible . I think yes. @plzen, what do You think about it?

Reply:
I've figured out that provider of cell is private and we can't use imitation of cell's selection so this implementation of test is correct in this case.